lib/iris/coord_categorisation.py

```diff
@@ -42,7 +51,11 @@ def add_categorised_coord(cube, name, from_coord, category_function, units="1"):
         coordinate point-value.
     units : str, default="1"
         Units of the category value, typically 'no_unit' or '1'.
-
+    category_function_expects_date : bool, default=False
```

Comment:
As an alternative to adding this extra argument, I could also add a new function `add_categorised_time_coord`. This function could then also take a simpler `category_function` argument, which would accept a function with only the point `value` argument, as the `coord` argument is no longer needed in any of the time categorization functions below.

Reply:
I have a third alternative: use `inspect.signature()` to check for type hinting on the second parameter. This would avoid needing `category_function_expects_date` or `add_categorised_time_coord`; it would all just work™. Obviously the docstring would need to make this behaviour clear. It seems a good way to open up the functionality for custom user coord categorisation, too.

`lambda`s don't properly support type hinting, so we'd have to abandon those, which I admit is a bit of a shame.

What do you think?

Reply:
Thanks for reviewing @trexfeathers! I implemented the proposed approach, could you have another look?